src/components/UniversalTable/UniversalTable.js

```diff
@@ -0,0 +1,55 @@
+import "./UniversalTable.css";
+
+export default function UniversalTable(collection, options) {
+  this.rowClick = options.onClick;
+  this.collection = collection;
+  this.parent = options.parent;
+  this.headerCells = options.header;
+}
+
+UniversalTable.prototype.render = function () {
+  const tableHeader = document.createElement("ul");
+
+  const tableHeaderRow = document.createElement("li");
+
+  tableHeaderRow.innerHTML = this.headerCells.reduce((acc, currentValue) => {
+    return acc + `<span>${currentValue}</span>`;
+  }, "");
+
+  tableHeader.append(tableHeaderRow);
+
+  const tableBody = document.createElement("ul");
+
+  if (!this.rowClick) {
+    console.log("No row click callback attached");
+  } else {
+    tableBody.onclick = (e) => {
+      this.rowClickHandler(e);
+    };
+  }
+
+  this.collection.forEach((row, index) => {
+    const tableRow = document.createElement("li");
+    tableRow.dataset.ind = index;
+    tableRow.classList.add("t-row");
+
+    tableRow.innerHTML = Object.values(row).reduce((acc, propertyValue) => {
```

Comment:
doesn't match the headers. so if somebody change the order of the headers it won't work correctly